File: 5KYU/Directions Reduction/solution.clj
(ns dir-reduc.core
  (:require [clojure.string :as str])) 

(defn reduce-directions [xs]
  ((fn [directions]
     (let [reduction (str/replace directions #"NORTHSOUTH|EASTWEST|SOUTHNORTH|WESTEAST" "")]
       (if (= reduction directions)
           (re-seq #"EAST|NORTH|WEST|SOUTH" reduction)
           (recur reduction)))) (apply str xs)))

File: 5KYU/Directions Reduction/solution.go
package kata

import (
  "strings"
  "regexp"
)

func DirReduc(arr []string) []string {
    directions := strings.Join(arr, "")
    for {
      beforeReplace := directions
      directions = strings.Replace(directions, "NORTHSOUTH", "", -1)
      directions = strings.Replace(directions, "SOUTHNORTH", "", -1)
      directions = strings.Replace(directions, "EASTWEST", "", -1)
      directions = strings.Replace(directions, "WESTEAST", "", -1)
      
      if beforeReplace == directions {
        if directions == "" {
          return []string{}
        }
        re := regexp.MustCompile("(NORTH|SOUTH|EAST|WEST)")
	      return re.FindAllString(directions, -1)        
      }
    }
  return []string{}
}